api/provider/gitlab.go

```diff
@@ -70,21 +82,23 @@ func (g gitlabProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*Us
 		},
 	}
 
-	var emails []*githubUserEmail
+	var emails []*gitlabUserEmail
 	if err := makeRequest(ctx, tok, g.Config, g.Host+"/api/v4/user/emails", &emails); err != nil {
 		return nil, err
 	}
 
-	if len(emails) > 0 {
-		data.Email = emails[0].Email
```

Comment:
This seems like a bug, GitLab `/api/v4/user/emails` API will return only the additional emails excluding the primary one (very strange indeed). As a result if a user has an additional GitLab email that one was used instead of the primary one like GitHub and Bitbucket

Reply:
Any Gitlab accounts we could double check?

Reply:
Tested with my GitLab account.
1. Created a site with GitLab auth provider.
2. Invited my Gmail address (primary email in GitLab).
3. Logged in via GitLab successfully.
4. Added my Netlify email to my GitLab account (unverified).
5. Tried to login again to my site via GitLab and failed with "Signups not allowed".
6. Invited my Netlify email but didn't click the invite link.
7. Tried to logic via GitLab - failed with no error but got a confirmation email (not an invite one). Clicking the confirmation link or the previous invite link doesn't fix the login issue.

To fix the login I could either re-invite my Netlify email via Identity and click the invite link or verify the email on GitLab.